plugins/inputs/dpdk/dpdk.go

```diff
@@ -0,0 +1,240 @@
+  device_types = ["ethdev", "rawdev"]
+
+  ## List of custom, application-specific telemetry commands to query 
+  ## For e.g. L3 Forwarding with Power Management Sample Application this could be: 
+  ##   additional_commands = ["/l3fwd-power/stats"]
+  # additional_commands = []
+
+  ## Allows turning off collecting data for individual "ethdev" commands.
+  ## Remove "/ethdev/link_status" from list to start getting link status metrics.
+  [inputs.dpdk.ethdev]
+    exclude_commands = ["/ethdev/link_status"]
+
+  ## When running multiple instances of the plugin it's recommended to add a unique tag to each instance to identify
+  ## metrics exposed by an instance of DPDK application. This is useful when multiple DPDK apps run on a single host.  
+  ##  [inputs.dpdk.tags]
+  ##    dpdk_instance = "my-fwd-app"
+`
+	defaultPathToSocket        = "/var/run/dpdk/rte/dpdk_telemetry.v2"
+	defaultAccessTimeout       = config.Duration(200 * time.Millisecond)
+	maxCommandLength           = 56
+	maxCommandLengthWithParams = 1024
+	dpdkPluginName             = "dpdk"
+	ethdevListCommand          = "/ethdev/list"
+	rawdevListCommand          = "/rawdev/list"
+)
+
+type dpdk struct {
+	SocketPath         string          `toml:"socket_path"`
+	AccessTimeout      config.Duration `toml:"socket_access_timeout"`
+	DeviceTypes        []string        `toml:"device_types"`
+	EthdevConfig       ethdevConfig    `toml:"ethdev"`
+	AdditionalCommands []string        `toml:"additional_commands"`
+	Log                telegraf.Logger `toml:"-"`
+	connector          *dpdkConnector  `toml:"-"`
+	rawdevCommands     []string        `toml:"-"`
+	ethdevCommands     []string        `toml:"-"`
+}
+
+type ethdevConfig struct {
+	EthdevExcludeCommands []string `toml:"exclude_commands"`
+}
+
+func init() {
+	inputs.Add(dpdkPluginName, func() telegraf.Input {
+		dpdk := &dpdk{
+			SocketPath:     defaultPathToSocket,
+			AccessTimeout:  defaultAccessTimeout,
+			rawdevCommands: []string{"/rawdev/xstats"},
+			ethdevCommands: []string{"/ethdev/stats", "/ethdev/xstats", "/ethdev/link_status"},
+		}
+		return dpdk
+	})
+}
+
+func (dpdk *dpdk) SampleConfig() string {
+	return dpdkSampleConfig
+}
+
+func (dpdk *dpdk) Description() string {
+	return dpdkDescription
+}
+
+// Performs validation of all parameters from configuration
+func (dpdk *dpdk) Init() error {
+	if dpdk.SocketPath == "" {
+		dpdk.SocketPath = defaultPathToSocket
+		dpdk.Log.Infof("using default '%v' path for socket_path", defaultPathToSocket)
+	}
+	if err := isSocket(dpdk.SocketPath); err != nil {
+		return err
+	}
+
+	if err := dpdk.validateCommands(); err != nil {
+		return err
+	}
+
+	if dpdk.AccessTimeout < 0 {
+		return fmt.Errorf("socket_access_timeout should be positive number or equal to 0 (to disable timeouts)")
+	}
+	if len(dpdk.AdditionalCommands) == 0 && len(dpdk.DeviceTypes) == 0 {
+		dpdk.Log.Warn("DPDK plugin is enabled and configured with nothing to read")
```

Comment:
It's a philosophical question, but does it really make sense to run a configuration like this?

Reply:
No, that's maybe why there is the warning? Users then can figure out why they don't receive any metrics.

Reply:
Exactly, this is why there is the warning. 
Alternatively, we can return error which will cause Telegraf to fail and exit. But I wouldn't do that for somehow valid configuration.
@srebhan ?

Reply:
To be honest, this is now the second or third plugin that allows this and I completely miss the use case. Why should I configure a plugin that will not report any metric in this (valid) config? This is with 99% certainty a configuration error. [Unix rule 12](https://homepage.cs.uri.edu/~thenry/resources/unix_art/ch01s06.html) says: "When you must fail, fail noisily and as soon as possible.". Now we can argue if we need to fail but having a plugin that produces nothing is not very helpful either....
Please enlighten me: What is your use-case to make this a valid config?

Reply:
@srebhan You're right. Will fail fast now.